File: src/25-challenge/solution.py
from typing import Union, Literal

type Instruction = Union[
    Literal["-"],
    Literal["+"],
    Literal[">"],
    Literal["["],
    Literal["]"],
    Literal["{"],
    Literal["}"],
]


def execute(code: str) -> int:
    result: int = 0

    for i, instruction in enumerate(iterable=code):
        match (instruction):
            case "+":
                result += 1
                break

            case "-":
                result -= 1
                break

            case "[":
                result = 0
                i: int = code.index("]", i)
                break

            case "{":
                if not result:
                    i: int = code.index("}", i)
                break

    return result

File: src/25-challenge/solution.ts
type Instruction = '-' | '+' | '>' | '[' | ']' | '{' | '}'

export default function execute(code: string): number {
	let result: number = 0

	for (let i = 0; i < code.length; i++) {
		const instruction: Instruction = code[i] as Instruction

		switch (instruction) {
			case '+':
				result++
				break

			case '-':
				result--
				break

			case '[':
				result = 0
				i = code.indexOf(']', i)
				break

			case '{':
				if (!result) i = code.indexOf('}', i)
				break
		}
	}

	return result
}
